src/compat.h

```diff
@@ -12,6 +12,13 @@
 #define PG96 is_supported_pg_version_96(PG_VERSION_NUM)
 #define PG10 is_supported_pg_version_10(PG_VERSION_NUM)
 
+#ifndef TupleDescAttr
+/*
+   backport in PG10 commit d34a74dd
+*/
+#define TupleDescAttr(a, i) ((a)->attrs[(i)])
+#endif
+
```

Comment:
This compat thingy is IMO not needed, because TupleDescAttr was backpatched to 9.2+.

Reply:
Hi,
Yes it is backpatched to 9.2, but I was doing git bisect and pg V9.6/V10/V11 fork is older than the backpatch. An always valid TupleDescAttr during the bisect is nice to have.

Reply:
I wouldn't personally expect that people build against other than reasonably new minor releases.